controller/src/main/java/io/pravega/controller/store/kvtable/AbstractKVTableMetadataStore.java

```diff
@@ -128,6 +128,13 @@ public KeyValueTable getKVTable(String scope, final String name, OperationContex
         return Futures.completeOn(checkScopeExists(scope, context, executor)
                 .thenCompose(exists -> {
                     if (exists) {
+                        checkScopeInDeletingTable(scope, context, executor)
+                                .thenCompose(exist -> {
+                                    if (exist) {
+                                        throw new IllegalArgumentException("Scope already in deleting state: " + scope);
```

Comment:
checkScopeExists is being used at several places for lots of API calls. Changing that will need a thorough fresh lookup on every places it is being used right now. May be fixing at one place would break at a different one. A very nice point though. I'll take up this as a separate task to see if a new method should be created to do both things or simply we can club them.

Reply:
it's handled now